src/Compression/ZipArchive.class.st

```diff
@@ -271,7 +271,7 @@ ZipArchive >> readFrom: aStreamOrFile [
 	| stream name eocdPosition |
 	stream := aStreamOrFile isStream
 		ifTrue: [ name := aStreamOrFile printString. aStreamOrFile ]
-		ifFalse: [ name := aStreamOrFile asFileReference fullName. aStreamOrFile asFileReference binaryReadStream ].
+		ifFalse: [ name := aStreamOrFile asFileReference fullName. aStreamOrFile asFileReference binaryReadStream contents readStream ].
```

Comment:
```suggestion
		ifFalse: [ 
			name := aStreamOrFile asFileReference fullName.
			
			"Workaround: Force load the entire contents in memory.
			The zip read stream breaks with Zn buffered streams otherwise"
			aStreamOrFile asFileReference binaryReadStream contents readStream ].
```